manila/api/v2/share_networks.py

```diff
@@ -397,7 +398,8 @@ def _consumed(name):
                     context, share_network_subnet_values)
             except db_exception.DBError:
                 db_api.share_network_delete(context, share_network['id'])
-                msg = _('Could not create share network.')
+                QUOTAS.rollback(context, reservations)
```

Comment:
Same like above (not just on `DBError`) and additionally any error on `share_network_delete` should not prevent the quota rollback, so put it in a try..finally block, please